src/index.ts

```diff
@@ -45,17 +46,85 @@ passport.use('local', new LocalStrategy(
       .then((doc: any) => {
         bcrypt.compare(password, doc.password, (err, correct) => {
           if (err) return done(err)
-          if (!correct) return done(null, false, { message: 'Incorrect password' })
+          if (!correct) return done(null, false, { message: _CC.lang('LOGIN_INCORRECT_PASSWORD') })
           if (correct) return done(null, doc)
         })
       })
       .catch(err => {
-        if (err.message === 'missing') return done(null, false, { message: 'Incorrect username.' })
+        if (err.message === 'missing') return done(null, false, { message: _CC.lang('LOGIN_INCORRECT_USERNAME') })
         return done(err)
-      })
+      })  
   }
 ))
 
+if ( config.googleSSOEnabled ) {
+  passport.use('google-login', new GoogleStrategy({
+      clientID: config.googleSSOClientId,
+      clientSecret: config.googleSSOClientSecret,
+      callbackURL: config.googleSignInRedirect,
+    },
+    async (issuer, profile, done) => {
+      const username = profile.emails[0].value.trim();
```

Comment:
This should use `profile.id` rather than the first email, because the emails associated with a Google account can change.

Reply:
Resolved with https://github.com/Wingysam/Christmas-Community/pull/138/commits/607b7887945799def64dee7415b2b739b82f4151